src/definitions/definitionBlocks.ts

```diff
@@ -58,13 +58,9 @@ export type NexusOutputFieldDef = NexusOutputFieldConfig<string, any> & {
 
 // prettier-ignore
 export type ScalarOutSpread<TypeName extends string, FieldName extends string> =
-  NeedsResolver<TypeName, FieldName> extends true
-    ? HasGen3<'argTypes', TypeName, FieldName> extends true
-      ? [ScalarOutConfig<TypeName, FieldName>]
-      : [ScalarOutConfig<TypeName, FieldName>] | [FieldResolver<TypeName, FieldName>]
-    : HasGen3<'argTypes', TypeName, FieldName> extends true
-      ? [ScalarOutConfig<TypeName, FieldName>]
-      : [] | [FieldResolver<TypeName, FieldName>] | [ScalarOutConfig<TypeName, FieldName>]
```

Comment:
Could you explain the reason why we no longer need the `NeedsResolver` type?

Reply:
Ah, I think we should be keeping that you're right. When `NeedsResolver` is true then `[]` should never be an option.

It seems we don't have any test coverage for this case... I'll see if I can add some.